### PR TITLE
.Net: Fix OpenAI JSON manifest parsing (underscore cases)

### PR DESCRIPTION
**Pull request by crickman**
### Motivation and Context

Enum values with underscore resulting in runtime exception when encountered:

```
Microsoft.SemanticKernel.Diagnostics.SKException: Parsing of Open AI manifest failed. ---> System.Text.Json.JsonException: 
The JSON value could not be converted to Microsoft.SemanticKernel.Functions.OpenAPI.OpenAI.OpenAIAuthenticationType. Path: $.type | LineNumber: 0 | BytePositionInLine: 19. 
at System.Text.Json.ThrowHelper.ThrowJsonException(String message) at System.Text.Json.Serialization.Converters.EnumConverter`1.ReadEnumUsingNamingPolicy(String enumString) 
at System.Text.Json.Serialization.Converters.EnumConverter`1.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options) 
at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader) 
at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value) 
at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value) 
at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state) at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 utf8Json, JsonTypeInfo jsonTypeInfo, Nullable`1 actualByteCount) 
at System.Text.Json.JsonSerializer.ReadNode[TValue](JsonNode node, JsonTypeInfo jsonTypeInfo) at System.Text.Json.JsonSerializer.Deserialize[TValue](JsonNode node, JsonSerializerOptions options) 
at Microsoft.SemanticKernel.Functions.OpenAPI.OpenAI.KernelOpenAIPluginExtensions.ImportAsync(IKernel kernel, String openAIManifest, String pluginName, OpenAIFunctionExecutionParameters executionParameters, CancellationToken cancellationToken)
```

### Description

The existing `JsonStringEnumConverter` attribution works fine for serialization (which we don't utilize), but insufficient for _deserialization_.  This is solved in .Net8 but needs a custom JsonNamingPolicy for `netstandard2.0`

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by TaoChenOSU — APPROVED** [review]

**Review by glahaye — APPROVED** [review]

File: dotnet/src/Functions/Functions.OpenAPI/OpenAI/SnakeCaseJsonNamingPolicy.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.SemanticKernel.Functions.OpenAPI.OpenAI;
+
+/// <summary>
+/// Available in .Net8:
+/// https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonnamingpolicy.snakecaselower?view=net-8.0#system-text-json-jsonnamingpolicy-snakecaselower
```
Comment (stephentoub) [inline dotnet/src/Functions/Functions.OpenAPI/OpenAI/SnakeCaseJsonNamingPolicy.cs:13]: You should be able to just use this now. We're now referencing the 8.0 version of System.Text.Json.

File: dotnet/src/Functions/Functions.OpenAPI/OpenAI/KernelOpenAIPluginExtensions.cs [hunk 2]
```diff
@@ -195,7 +202,7 @@ public static class KernelOpenAIPluginExtensions
         try
         {
             pluginJson = JsonNode.Parse(openAIManifest)!;
-            openAIAuthConfig = pluginJson["auth"].Deserialize<OpenAIAuthenticationConfig>()!;
+            openAIAuthConfig = pluginJson["auth"].Deserialize<OpenAIAuthenticationConfig>(s_jsonOptionsOpenAIManifest)!;
```
Comment (stephentoub) [inline dotnet/src/Functions/Functions.OpenAPI/OpenAI/KernelOpenAIPluginExtensions.cs:205]: I'm a little confused... which members is this snake casing meant to influence?  Most of the members I see have an explicit JsonPropertyName, and this change is also removing an enum and replacing it with a string.

Comment (crickman) [reply dotnet/src/Functions/Functions.OpenAPI/OpenAI/KernelOpenAIPluginExtensions.cs:205]: Thanks for this question Stephen.  The high-order bit here is that our OpenAI function calling story is functionally broken.  This is a priority fix.

This issue at play has to do with the enums: `OpenAIAuthenticationType` and `OpenAIAuthorizationType`.  This is where the snake casing on deserialization was causing issue (re: stack trace).

The reason I removed the third enum is that is just seems silly to define mime/type as enum and then attempt to marshall value conversion...when mime-type is normally adequately handled as string.  Also, if a new mime-type is added...it won't break enum parsing.  (Apparently EnumMemberAttribute doesn't impact json deserialization and this code path was never tested.)

Comment (stephentoub) [reply dotnet/src/Functions/Functions.OpenAPI/OpenAI/KernelOpenAIPluginExtensions.cs:205]: > Apparently EnumMemberAttribute doesn't impact json deserialization

It doesn't impact JsonSerializer, period :) It's part of an entirely different serializer.  All usage here should be removed.

**Review by stephentoub — APPROVED** [review]

**Review by alliscode — APPROVED** [review]

**Pull request merged**